content/module-1/6-messaging-protocol.md

```diff
@@ -9,20 +9,16 @@ The Topos protocol provides for the creation, verification and delivery of certi
 
 The developers of dApps and cross-subnet messaging flows can leverage these certificates to design **secure messaging protocols**. By storing relevant certificates on-chain, it is possible to benefit from their security guarantees when handling cross-subnet messages.
 
-To demonstrate best practice and help bootstrap subnet administrators, Topos has created a series of smart contracts:
+To demonstrate best practices and help bootstrap subnet administrators, Topos has created a series of smart contracts:
 
 1. The **ToposCore** smart contract can be used to store certificates on-chain. Although not compulsory, it makes sense for a subnet to have a single instance of such a certificate-storing smart contract.
-2. The **ToposMessaging** smart contract makes use of ToposCore. It has functions to verify proofs of inclusion and to mark completed messages as used, thereby preventing replay. It is not meant to be deployed directly, but instead to be used as an inherited smart contract for the messaging protocol proper.
+2. The **ToposMessaging** smart contract gathers functionalities that are core to messaging protocols. It has functions to verify proofs of inclusion and to mark completed messages as used, thereby preventing replay. To ease the development of new messaging protocols, these functionalities can be obtained for free by creating a messaging protocol smart contract that inherits ToposMessaging.
```

Comment:
Would it be fair to say that this is part of a different piece work, a rework of B9's content?

This PR contains changes in the first module because a few things were unclear. Now, adding things that make sense like what you said would require re-working a greater piece of the content, notably the way we introduce messaging protocols in the whole portal, which sounds like something much bigger than a simple PR comment suggestion application, no?

Reply:
That is a fair POV. I think it would be reasonable to create a ticket to follow _this_ PR up with some additional work that clarifies those other areas more.

Otherwise, yes, there is the risk that the scope creeps too much on this set of changes. I'd rather see the content live soon, even if there are things that can be adjusted and improved later.